app/controllers/media_controller.rb

```diff
@@ -19,18 +19,29 @@ def stream
     authorize! :read, @media
     respond_to do |format|
       format.m3u8 do
-        redirect_to @media.to_playlist_url
+        redirect_to "#{@media.to_playlist_url}?token_string=#{media_token}"
```

Comment:
Since `@media.to_playlist_url` is a string URL to an external resource (and not a hash of params to an internal resource) I can't think of a more elegant solution off the top of my head.

More than happy to look at potential alternative solutions.